src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -232,7 +235,7 @@ public bool TryGetValue(object key, out object result)
                     entry.LastAccessed = utcNow;
                     result = entry.Value;
 
-                    if (entry.CanPropagateOptions())
+                    if (_options.TrackLinkedCacheEntries && entry.CanPropagateOptions())
```

Comment:
Should use the "stored setting value", so it is consistent for the entire MemoryCache lifetime. 

```suggestion
                    if (TrackLinkedCacheEntries && entry.CanPropagateOptions())
```